ibis/backends/tests/test_generic.py

```diff
@@ -254,3 +255,72 @@ def test_select_filter_mutate(backend, alltypes, df):
     expected = expected.assign(float_col=expected['float_col'].astype('int32'))
 
     backend.assert_frame_equal(result, expected)
+
+
+def test_fillna_invalid(alltypes):
+    with pytest.raises(
+        com.IbisTypeError, match=r"value \['invalid_col'\] is not a field in.*"
+    ):
+        alltypes.fillna({'invalid_col': 0.0})
+
+
+def test_dropna_invalid(alltypes):
+    with pytest.raises(
+        com.IbisTypeError, match=r"value 'invalid_col' is not a field in.*"
+    ):
+        alltypes.dropna(subset=['invalid_col'])
+
+    with pytest.raises(ValueError, match=r".*is not in.*"):
+        alltypes.dropna(how='invalid')
+
+
+@pytest.mark.parametrize(
+    'replacements',
+    [
+        0.0,
+        0,
+        1,
+        ({'na_col': 0.0}),
+        ({'na_col': 1}),
+        ({'none_col': 0.0}),
+        ({'none_col': 1}),
+    ],
+)
+@pytest.mark.only_on_backends(['pandas', 'dask', 'pyspark'])
+def test_fillna_table(backend, alltypes, replacements):
+    table = alltypes.mutate(na_col=np.nan)
+    table = table.mutate(none_col=None)
+    table = table.mutate(none_col=table['none_col'].cast('float64'))
+    table_pandas = table.execute()
+
+    result = table.fillna(replacements).execute().reset_index(drop=True)
+    expected = table_pandas.fillna(replacements).reset_index(drop=True)
+
+    backend.assert_frame_equal(result, expected, check_dtype=False)
+
+
+@pytest.mark.parametrize(
+    ('how', 'subset'),
+    [
+        ('any', None),
+        ('any', []),
+        ('any', ['int_col', 'na_col']),
+        ('all', None),
+        ('all', ['int_col', 'na_col']),
+        ('all', ['none_col']),
+    ],
+)
+@pytest.mark.only_on_backends(['pandas', 'dask', 'pyspark'])
+def test_dropna_table(backend, alltypes, how, subset):
+    table = alltypes.mutate(na_col=np.nan)
+    table = table.mutate(none_col=None)
+    table = table.mutate(none_col=table['none_col'].cast('float64'))
+    table_pandas = table.execute()
+
+    result = table.dropna(subset, how).execute().reset_index(drop=True)
+    subset = subset if subset else table_pandas.columns
+    expected = table_pandas.dropna(how=how, subset=subset).reset_index(
+        drop=True
+    )
+
+    backend.assert_frame_equal(result, expected, check_dtype=False)
```

Comment:
why do you have `check_type=False`? this should match

Reply:
So when the CI ran for this, the pyspark tests with Java 8 had diffs in the column type when doing fillna(int_value) on a null column (int in pandas vs. float in Pyspark), and also differences in the column type of a bool_col of an empty DataFrame (bool vs. object): https://github.com/ibis-project/ibis/runs/4051801741?check_suite_focus=true
The pyspark tests on Java 11 passed fine though.

Reply:
kk i have to loook at the doc again, but we should highlite this in the Notes section (e.g. potential lack of type stability) on various backends (also add a comment here)